R/as_noise_params.R

```diff
@@ -0,0 +1,78 @@
+#' Load noise parameters
+#'
+#' @param data a [data.frame()] or [tibble::tbl_df]
+#' @param antigen_isos [character()] vector of antigen isotypes
+#' to be used in analyses
+#'
+#' @returns a `noise_params` object (a [tibble::tbl_df] with
+#' extra attribute `antigen_isos`)
+#' @export
+#' @examples
+#' library(magrittr)
+#' noise_data <-
+#'   "https://osf.io/download//hqy4v/" %>%
+#'   readr::read_rds() %>%
+#'   as_noise_params()
+#'
+#' print(noise_data)
+#'
+as_noise_params <- function(data, antigen_isos = NULL) {
+  if (!is.data.frame(data)) {
+    cli::cli_abort(
+      class = "not data.frame",
+      message = c(
+        "Can't convert {.arg data} to {.cls noise_params}.",
+        paste("x" = "{.arg data} must be a {.cls data.frame}",
+          "(or a subclass of {.cls data.frame})."
+        ),
+        "i" = "You have supplied a {.cls {class(data)}}."
+      )
+    )
+  }
+
+  noise_data <- data %>% tibble::as_tibble()
+
+  # Define noise columns
+  noise_cols <- c("antigen_iso", "y.low", "eps", "nu", "y.high")
+
+  # Get any missing columns
+  missing_cols <- setdiff(x = noise_cols, y = names(data))
+
+  if (length(missing_cols) > 0) {
+    cli::cli_abort(
+      class = "not noise_params",
+      message = c(
+        "Can't convert {.arg data} to {.cls noise_params}.",
+        "i" = paste("The column{?s} : {.strong {.var {missing_cols}}}",
```

Comment:
there's an extra space **before** the colon here (keep the one **after** the colon)

Reply:
This is well-noted. Do note that the `styleR` library does not put spaces before or after the colon.

```
  if (length(missing_cols) > 0) {
    cli::cli_abort(
      class = "not noise_params",
      message = c(
        "Can't convert {.arg data} to {.cls noise_params}.",
        "i" = paste(
          "The column{?s}:{.strong {.var {missing_cols}}}",
          "{?is/are} missing."
        )
      )
    )
  }
   ```

Reply:
Does `styler` even attempt to edit strings? I'd guess not.
Even if it does - it's fine to use `styler` or other tools to help clean up code, but please still do a final, detailed visual edit before requesting review.

Reply:
`styleR` does edit strings. The space before the colon has been removed.